src/dlgConnectionProfiles.cpp

```diff
@@ -252,6 +259,7 @@ void dlgConnectionProfiles::slot_update_url(const QString& url)
 {
     if (url.isEmpty()) {
         validUrl = false;
+        offline_button->setDisabled(true);
```

Comment:
There is a code path where a pHost is connected with no checks if it exists. Having not traced the entire flow of the program, I am not certain that creating a profile with no url or port would not cause issues some other way. So I thought it best to require the same level of initialization for any selected or created profile that comes out of this dialog, regardless of whether the user connects it immediately or not.
It's not a difficult change to make, but I would not be comfortable submitting it in this PR unless there is a more robust set of regression test than I am aware of.
Incidentally, I'm working on that.

Reply:
Yeah - I can see that it ***is*** simpler to require a properly documented connection even if there is no immediate intention to use it - because the complication of *subsequently* trying to permit/deny a connection being attempted *without even having the connection dialogue active/open* is going to make this PR more complex than perhaps it is already.

OTOH a later PR to enhance things so that a profile can be created without having all the details to hand might be easier for some atypical users that use one or more profiles for other things.  And they can always do what I do currently if I want to open a profile without making a connection and that is to specify `localhost` as the **Server:** (and set an unused **Port:**, or even the default `23`) - as I am not running a *telnet* server locally that means the connection does not ever do anything... {since we still do not have code that makes use of `(int) Host::mTimeOut` or `(int) Host::mRetries` :stuck_out_tongue_closed_eyes: }